proposals/XXXX-compact.md

```diff
@@ -0,0 +1,280 @@
+When features of Compact are similar or analogous to features of TypeScript, we have used the same syntax.
+When features are *different*, we have intentionally used different syntax.
+
+## Safety
+The execution of smart contracts can have real-world financial consequences.
+The addition of data privacy can also raise issues of legal requirements of data protection on the one hand and data disclosure on the other.
+While Compact itself cannot solve these issues, it is intended to make it easy for contract authors to address them.
+It is therefore important that it is easy to develop correct smart contracts,
+and possible to completely understand their behavior by inspecting their source code.
+
+Compact is statically typed to help enforce correctness.
+The Compact type system is stronger than TypeScript's.
+There is no `any` type,
+type casts are checked at runtime when necessary rather than simply trusted,
+and the boundary with external code contains runtime type checks to protect the Compact code.
+
+Compact has a very clear distinction between private data (called "witnesses" in Compact) and public data (called the "ledger" in Compact).
+The language's implementation tracks private data and requires it to be explicitly disclosed by the developer before allowing it to be publicly observed.
+Failure to explicitly disclose such private data is a compilation error, preventing the contract from ever being executed.
+
+## Restriction
+The language is not a general purpose language, it is intentionally restricted.
+In the context of a privacy-preserving blockchain, these restrictions come from two different sources.
+First, there is an underlying zero-knowledge proving system, which will have some restrictions on the data types and computations it can represent.
+Second, there is an intended blockchain which will have restrictions on the work that can be done as part of a transaction, depending on characteristics such as block size, consensus model, etc.
+Blockchain restrictions can also be required in order to accurately charge fees for transactions and to prevent denial of service attacks on the blockchain.
+
+The current implementation of Compact has the specific restrictions of the Midnight Network blockchain and its underlying BLS proving system.
+However, Compact itself is intended to be able to use "pragmas" in a contract's implementation to choose different restrictions.
+
+# Status
+The status of Compact is [incubation](https://toc.hyperledger.org/governing-documents/project-lifecycle.html#incubation)
+
+# Solution
+Compact is a language that is designed to support a range of privacy-preserving smart contract systems.
+It has both an informal language reference and a formal specification in Agda.
+Both of these specifications are in progress and evolving as the language changes.
+There is also an official implementation in the form of the Compact compiler, and a suite of developer tools forming a Compact software development kit (SDK).
```

Comment:
Are both of these implementations (compiler and SDK) being proposed for inclusion in this project?

Reply:
Yes.  They are both part of the Compact open-source (not yet public) repository that we are preparing.